### DISCUSSION TITLE
Equivalent for `readFragment` for retrieving node data from cache

### DISCUSSION THREAD

**Discussion by momakes2** [urql-graphql/urql#1338] [Q&A] [answered]
Hey! Do you think you'll provide a "readFragment" exposed via useClient? 

My use case is having an abstraction to read from the urql cache but using an API like: 

```getUserById(id:string): UserFragmentInfo```

but I see why you wouldn't provide that as this result could be not cached. 

What do you think is a good way to achieve this so:
1. we minimize queries to the server because 99% of the time a `User` is cached via a prior query.
2. we provide fast access for quick lookup functions like `getUserById`

**Comment by JoviDeCroock** [comment, marked as answer]
Hey,

If you are using graphCache this is possible through the resolvers api. When we have a user in cache through a list query,... we can hook up the details query like this https://formidable.com/open-source/urql/docs/graphcache/computed-queries/#resolve (scroll down to the todo key)

We have had an RFC for a relay-like `useFragment` but ended up deciding against it.

We chose not to attach it to the client because this won't work with a document-cache.

**Reply by momakes2** [reply]
Thanks for the swift response. How can I access a User entity outside of update resolvers? I want to look up a user in my `getUserById` function in my custom hook.

**Reply by momakes2** [reply]
Oh saw it when you edited! Thank you, I don't think if I paid a person, I'd get such a fast response. <3
